sql/plan/indexed_table_access.go

```diff
@@ -127,6 +126,96 @@ func (i *IndexedTableAccess) RowIter(ctx *sql.Context, row sql.Row) (sql.RowIter
 	return sql.NewSpanIter(span, sql.NewTableRowIter(ctx, indexedTable, partIter)), nil
 }
 
+type LookupBuilder struct {
+	keyExprs        []sql.Expression
+	keyExprs2       []sql.Expression2
+	matchesNullMask []bool
+	index           sql.Index
+}
+
+func NewLookupBuilder(index sql.Index, keyExprs []sql.Expression, matchesNullMask []bool) LookupBuilder {
+	return LookupBuilder{
+		index:           index,
+		keyExprs:        keyExprs,
+		matchesNullMask: matchesNullMask,
+	}
+}
+
+func (lb LookupBuilder) GetLookup(ctx *sql.Context, key LookupBuilderKey) (sql.IndexLookup, error) {
+	ib := sql.NewIndexBuilder(ctx, lb.index)
+	iexprs := lb.index.Expressions()
+	for i := range key {
+		if key[i] == nil && lb.matchesNullMask[i] {
+			ib = ib.IsNull(ctx, iexprs[i])
+		} else {
+			ib = ib.Equals(ctx, iexprs[i], key[i])
```

Comment:
Probably want to push this logic into the analyzer (it's a bit jank how we assume an equality relationship here when that's not part of the contract of a lookup).

Add a note for now, and make sure there's a test that we don't try to do this for e.g. `join t2 on a > b` or something. We might already have that test, but I'm not sure